cmake/Embed.cmake

```diff
@@ -140,10 +140,13 @@ function(add_embed_library EMBED_NAME)
     endforeach()
     message(STATUS "Generating embedding library '${EMBED_NAME}'")
     generate_embed_source(${EMBED_NAME} ${SRC_FILE} ${HEADER_FILE} "${PARSE_BASE_DIRECTORY}" SYMBOLS ${SYMBOLS} FILES ${PARSE_UNPARSED_ARGUMENTS})
-    add_library(object_${EMBED_NAME} OBJECT ${SRC_FILE} ${HEADER_FILE})
+    add_library(object_${EMBED_NAME} OBJECT ${SRC_FILE} ${HEADER_FILE} $<$<NOT:$<BOOL:${EMBED_USE_LD}>>:${OUTPUT_FILES}>)
```

Comment:
Use an `if` statement. `EMBED_USE_LD` is a configure-time variable so there is no need to use generator expressions. Plus, the `if` is much more readable.

Reply:
We need generator expressions because of MSBuild, which is a multi-config generator.

Reply:
But you aren't using any of the `$<CONFIG>` expressions so its the same value across all configs so it can be set using `if` statements.